syntax = "proto3";

package minidfs;

//service methods for authentication
service AuthService {
    // login user and get token
    rpc Login(LoginReq) returns (AuthRes);

    // register user and get token
    rpc Register(RegisterReq) returns (AuthRes);
}

message LoginReq {
    string email = 1;
    string password = 2;
}

message RegisterReq {
    string email = 1;
    string password = 2;
}

message AuthRes {
    bool success = 1;
    string token = 2;
    string error_msg = 3;
}